src/types.rs

```diff
@@ -167,14 +167,22 @@ where
 impl<ChannelId, Pk, AccountId> Withdrawal<ChannelId, Pk, AccountId>
 where
 	Withdrawal<ChannelId, Pk, AccountId>: Encode,
-	Pk: IdentifyAccount<AccountId = Pk>,
+	Pk: IdentifyAccount<AccountId = Pk> + Clone,
+	ChannelId: Clone,
 {
 	/// Returns whether `sig` is a valid signature for this Withdrawal
 	/// and was created by the participant that wants to claim the funds.
 	pub fn validate_sig<Sig: Verify<Signer = Pk>>(&self, sig: &Sig) -> bool {
 		let msg = Encode::encode(&self);
 		sig.verify(&*msg, &self.part)
 	}
+
+	pub fn funding(&self) -> Funding<ChannelId, Pk> {
```

Comment:
You are converting a withdrawal into a funding? That sounds odd.

Reply:
Hm yes the naming is not intuitive. The `Funding` is only used to calculate the ID, i will think of a better name.